### PR TITLE
walletkit: add ListSweepsVerbose method to fetch verbose tx details

### PR DESCRIPTION
**Pull request by bhandras**
This commit adds a new method to the walletkit client to allow fetching sweeps with rich details as provided by LND.

#### Pull Request Checklist

- [x] PR is opened against correct version branch.
- [ ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [x] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

### CODE REVIEW

File: lnd_services.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ var (
 	minimalCompatibleVersion = &verrpc.Version{
 		AppMajor:  0,
 		AppMinor:  15,
-		AppPatch:  4,
+		AppPatch:  5,
```
Comment (guggero) [inline lnd_services.go:45]: Do we need to bump the minimal compatible version here? It looks to me like we only need the compile time dependency for the output script types, but the RPC call should be available in earlier versions as well?

File: walletkit_client.go [hunk 2]
```diff
@@ -424,6 +434,153 @@ func (m *walletKitClient) ListSweeps(ctx context.Context) ([]string, error) {
...
+		&walletrpc.ListSweepsRequest{
+			Verbose: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []lnwallet.TransactionDetail
+	// Since we have requested the abbreviated response from lnd, we can
```
Comment (guggero) [inline walletkit_client.go:566]: nit: this comment doesn't match the actual code (seems to be copied from the non-verbose version).

File: walletkit_client.go [hunk 3]
```diff
@@ -424,6 +434,153 @@ func (m *walletKitClient) ListSweeps(ctx context.Context) ([]string, error) {
...
+	// just get our response to a list of sweeps and return it.
+	rpcDetails := resp.GetTransactionDetails()
+	if rpcDetails != nil {
```
Comment (guggero) [inline walletkit_client.go:569]: Should we error out if this is nil? Since we explicitly requested the verbose version.

**Review by guggero — COMMENTED** [review]
Nice :tada:

File: lnd_services.go (hunk 1 above)
Comment (bhandras) [reply lnd_services.go]: Yes indeed, kept it at 0.15.4.

File: walletkit_client.go (hunk 2 above)
Comment (bhandras) [reply walletkit_client.go]: Ah yes, thanks! Fixed.

**Review by guggero — APPROVED** [review]

**Pull request merged**